src/main/java/net/greatstart/controllers/ProjectController.java

```diff
@@ -1,160 +1,105 @@
+        DtoProject project = projectService.getDtoProjectById(id);
+        if (project != null) {
+            return new ResponseEntity<>(project, HttpStatus.OK);
         }
         return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
 
-    @GetMapping(value = "/new")
-    public ModelAndView getAddProjectForm() {
-        ModelAndView model = new ModelAndView("project/add_project");
-        model.addObject(PROJECT, new DtoProject());
-        return model;
-    }
-
-    @PostMapping(value = "/new")
-    public ModelAndView addProject(@Valid DtoProject dtoProject,
-                                   Errors errors,
-                                   Principal principal,
-                                   @RequestParam(value = "file", required = false) MultipartFile file) {
-        if (errors.hasErrors()) {
-            return new ModelAndView("project/add_project");
+    @Transactional
+    @PreAuthorize("isAuthenticated()")
+    @PutMapping("/api/project/{id}")
+    public ResponseEntity<DtoProject> updateProject(
+            @PathVariable("id") long id,
+            @Valid @RequestBody DtoProject dtoProject) {
+        if (userService.getLoggedInUser().getId() != dtoProject.getOwner().getId()) {
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
         }
-        saveImage(dtoProject, file);
-        User owner = userService.getUserByEmail(principal.getName());
-        Project project = projectMapper.projectFromDto(dtoProject);
-        project.setOwner(owner);
-        projectService.saveProject(project);
-        return new ModelAndView(REDIRECT_TO_PROJECTS);
-    }
-
-    @GetMapping(value = "/{id}/update")
-    public ModelAndView getUpdateProjectForm(@PathVariable("id") Long id) {
-        if (id > 0) {
-            ModelAndView model = new ModelAndView("project/update_project");
-            Project project = projectService.getProjectById(id);
-            model.addObject(PROJECT, projectMapper.fromProjectToDto(project));
-            return model;
+        DtoProject currentProject = projectService.saveDtoProject(dtoProject);
+        if (currentProject != null) {
+            return new ResponseEntity<>(dtoProject, HttpStatus.OK);
         }
-        return new ModelAndView(PROJECTS);
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+
     }
 
-    @PostMapping(value = "/{id}/update")
-    public ModelAndView updateProject(@PathVariable Long id, @Valid DtoProject dtoProject, Errors errors,
-                                      @RequestParam(value = "file", required = false) MultipartFile file) {
-        if (errors.hasErrors()) {
-            return new ModelAndView("project/update_project");
+    @Transactional
+    @PreAuthorize("isAuthenticated()")
+    @PostMapping({"/api/project", "/api/project/"})
+    public ResponseEntity<DtoProject> newProject(
+            @Valid @RequestBody DtoProject dtoProject) {
+        dtoProject = projectService.createNewProjectFromDto(dtoProject);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "dtoProject". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)